### PR TITLE
Explain env vs envFrom for page Define Environment Variables for a Container

### PR DESCRIPTION
**Pull request by KhushPatibandha**
I resolved Issue #39873 “Define Environment Variables for a Container” does not explain env vs envFrom #39873

### CODE REVIEW

**Comment by sftim** [comment]
/retitleExplain env vs envFrom for page Define Environment Variables for a Container

File: content/en/docs/tasks/inject-data-application/define-environment-variable-container.md [hunk 1]
```diff
@@ -21,6 +21,14 @@ When you create a Pod, you can set environment variables for the containers
 that run in the Pod. To set environment variables, include the `env` or
 `envFrom` field in the configuration file.
 
+Now lets look at the difference between `env` and `envFrom`
```
Comment (sftim) [inline content/en/docs/tasks/inject-data-application/define-environment-variable-container.md:24]:
```suggestion
The `env` and `envFrom` fields have different effects.
```

File: content/en/docs/tasks/inject-data-application/define-environment-variable-container.md [hunk 2]
```diff
@@ -21,6 +21,14 @@ When you create a Pod, you can set environment variables for the containers
...
+
+`env` : This is a field in the pod specification that allows you to set environment variables for a specific container.
+
+`envFrom` :  This is a field in the pod specification that allows you to set environment variables by referencing a ConfigMap or a Secret. When you use `envFrom`, all the key-value pairs in the referenced ConfigMap or Secret are set as environment variables in the container. Hence `envFrom` is used to set multiple environment variables by referencing a ConfigMap or Secret
+
+You can read more about <a href="https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#configure-all-key-value-pairs-in-a-configmap-as-container-environment-variables"> ConfigMap </a> and <a href="https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#configure-all-key-value-pairs-in-a-secret-as-container-environment-variables"> Secret </a>
```
Comment (sftim) [inline content/en/docs/tasks/inject-data-application/define-environment-variable-container.md:30]: [Use Markdown, not HTML](https://kubernetes.io/docs/contribute/style/style-guide/#links), for links.

File: content/en/docs/tasks/inject-data-application/define-environment-variable-container.md [hunk 3]
```diff
@@ -21,6 +21,14 @@ When you create a Pod, you can set environment variables for the containers
 that run in the Pod. To set environment variables, include the `env` or
 `envFrom` field in the configuration file.
 
+Now lets look at the difference between `env` and `envFrom`
+
+`env` : This is a field in the pod specification that allows you to set environment variables for a specific container.
+
+`envFrom` :  This is a field in the pod specification that allows you to set environment variables by referencing a ConfigMap or a Secret. When you use `envFrom`, all the key-value pairs in the referenced ConfigMap or Secret are set as environment variables in the container. Hence `envFrom` is used to set multiple environment variables by referencing a ConfigMap or Secret
```
Comment (sftim) [inline content/en/docs/tasks/inject-data-application/define-environment-variable-container.md:28]:
```suggestion
`env`
: allows you to set environment variables for a container, specifying a value directly for each variable that you name.

`envFrom`
: allows you to set environment variables for a container by referencing either a ConfigMap or a Secret. When you use `envFrom`, all the key-value pairs in the referenced ConfigMap or Secret are set as environment variables for the container. You can also specify a common prefix string.

This page explains how to use `env`.
```

**Review by sftim — COMMENTED** [review]
Thanks. Here's some feedback.

**Comment by KhushPatibandha** [comment]
Thanks for the feedback I'll surely look into it.

**Comment by KhushPatibandha** [comment]
image
How about this? can I go forward with this?

**Comment by KhushPatibandha** [comment]
image
How about this? Can I go forward with this?

**Comment by sftim** [comment]
> Can I go forward with this? 

I recommend incorporating all the feedback. You're welcome to ask questions if you're not sure why reviewers have made a suggestion.

**Comment by KhushPatibandha** [comment]
I did, I changed the way definition were written as per you suggestion and I also used Markdown instead HTML for the hyperlinks. Do let me know if I am still missing out something.

File: content/en/docs/tasks/inject-data-application/define-environment-variable-container.md (hunk 1 above)
Comment (KhushPatibandha) [reply content/en/docs/tasks/inject-data-application/define-environment-variable-container.md]: suggested changes done.

File: content/en/docs/tasks/inject-data-application/define-environment-variable-container.md (hunk 2 above)
Comment (KhushPatibandha) [reply content/en/docs/tasks/inject-data-application/define-environment-variable-container.md]: suggested changes done.

File: content/en/docs/tasks/inject-data-application/define-environment-variable-container.md (hunk 3 above)
Comment (KhushPatibandha) [reply content/en/docs/tasks/inject-data-application/define-environment-variable-container.md]: suggested changes done.

File: content/en/docs/tasks/inject-data-application/define-environment-variable-container.md [hunk 4]
```diff
@@ -21,6 +21,18 @@ When you create a Pod, you can set environment variables for the containers
 that run in the Pod. To set environment variables, include the `env` or
 `envFrom` field in the configuration file.
 
+The `env` and `envFrom` fields have different effects.
+
+`env`
+: allows you to set environment variables for a container, specifying a value directly for each variable that you name.
+
+`envFrom`
+: allows you to set environment variables for a container by referencing either a ConfigMap or a Secret. When you use `envFrom`, all the key-value pairs in the referenced ConfigMap or Secret are set as environment variables for the container. You can also specify a common prefix string.
```
Comment (tengqm) [inline content/en/docs/tasks/inject-data-application/define-environment-variable-container.md:30]:
```suggestion
: allows you to set environment variables for a container by referencing either a ConfigMap or a Secret.
  When you use `envFrom`, all the key-value pairs in the referenced ConfigMap or Secret
  are set as environment variables for the container.
  You can also specify a common prefix string.
```

File: content/en/docs/tasks/inject-data-application/define-environment-variable-container.md [hunk 5]
```diff
@@ -21,6 +21,18 @@ When you create a Pod, you can set environment variables for the containers
...
+
+You can read more about [ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#configure-all-key-value-pairs-in-a-configmap-as-container-environment-variables) and [Secret](https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#configure-all-key-value-pairs-in-a-secret-as-container-environment-variables)
```
Comment (tengqm) [inline content/en/docs/tasks/inject-data-application/define-environment-variable-container.md:32]:
```suggestion
You can read more about [ConfigMap](/docs/tasks/configure-pod-container/configure-pod-configmap/#configure-all-key-value-pairs-in-a-configmap-as-container-environment-variables)
and [Secret](/docs/tasks/inject-data-application/distribute-credentials-secure/#configure-all-key-value-pairs-in-a-secret-as-container-environment-variables).
```

File: content/en/docs/tasks/inject-data-application/define-environment-variable-container.md (hunk 4 above)
Comment (KhushPatibandha) [reply content/en/docs/tasks/inject-data-application/define-environment-variable-container.md:30]: suggested changes done.

**Comment by tengqm** [comment]
/lgtm
Preview looks good.

**Review by sftim — COMMENTED** [review]
/label tide/merge-method-squash
/lgtm

**Comment by sftim** [comment]
> @sftim One rookie question, I shouldn't update any branch once "lgtm" is added right?

You can, but you'll need a new LGTM (we don't charge extra :wink:)

If you know how to squash down to 1 commit, please do that. No worries if you don't.

**Comment by KhushPatibandha** [comment]
Hey @sftim what should be my next steps to get this PR merged?

**Comment by nate-double-u** [comment]
> Hey @sftim what should be my next steps to get this PR merged?

Hi @KhushPatibandha, thanks for this contribution! (I see it's your first, welcome!) Sometimes PR are slow to get approval, it's not that they aren't good, or that there is something else that needs to be done, sometimes it's just that no-one has gotten to it.

But this looks good to me, so lets get it  merged in.

/approve

**Pull request merged**